src/utils/index.ts

```diff
@@ -37,11 +38,16 @@ export const getEquivalentResource = <Y, D>(
   currentResources: D[],
 ) => {
   if (resource) {
-    const [key, _] = Object.entries(resources ?? []).find((a) => isEqual(a[1], resource))!;
+    const [key, value] = Object.entries(resources ?? []).find((a) => _.isEqual(a[1], resource))!;
     return currentResources.find((e: D) => (e as any).stackResourceId === getResourceID(getStackName(context), key));
   }
 };
 
+export const removeNils = (o: object): object => {
+  // The conversion is a neat hack that removes nested `undefined` values
+  return JSON.parse(JSON.stringify(sanitize(o)));
```

Comment:
Do you really need the parse/stringify, given sanitize is already recursively going through the data structure and removing anything empty-ish?

Reply:
This was my paranoia kicking in 😅

Reply:
Hehe I fear that going to json and back may bring in more issues than it solves